1.1.1 backports

gsmet opened the pull request:
Please don't merge, I'll do it myself once everything is ready.

manovotn committed:
Provided ManagedExecutor and ThreadContext beans should be default so that users can override them easily.

loicmathieu committed:
Fix a small documentation error

paschi committed:
Fix file system path for overriding application.properties when using dev-mode with Gradle

gastaldi committed:
Minor cleanup in Gradle plugin

Using constants and avoiding casts

viniciusfcf committed:
auth-mechanism property does not exist

auth-mechanism property does not exist

geoand committed:
Fix broken call-out

geoand committed:
Make application version and name available as runtime properties

Fixes: #6255

Co-authored-by: Guillaume Smet <guillaume.smet@gmail.com>

ia3andy committed:
Fix table design on safari and mobiles

ia3andy committed:
Use uppercase only in placeholder for the config search

mejlholm committed:
Mention possibility to disable JSON for non prod

Remove <build> surrounding extension - it's not valid.
Add small example of how to use human readable logging for dev/test and structured JSON for production.

jaikiran committed:
issue-5942 Introduce a deterministic ordering, between extensions, for database schema updates

jaikiran committed:
Prevent UnsupportedOperationException in addSourcePaths

pivovarit committed:
Update writing-extensions.adoc

geoand committed:
Add tests for application name and version handling

michael-simons committed:
Fix package name of Neo4j health check.

ia3andy committed:
Add note about Sentry re-licensing to BSL

geoand committed:
Ensure that NativeIT is only run during when using native profile

The other tests are only run when this profile is not active in order
to avoid running them multiple times (since they are slow as well)

Co-authored-by: Guillaume Smet <guillaume.smet@gmail.com>

jaikiran committed:
issue-6280 Address Class-Path parsing issues with Java 11 and Windows, for tests launched using QuarkusDevModeTest

maxandersen committed:
sentry.io is not opensource

gsmet committed:
Only have one warning about the non-open source Sentry license

geoand committed:
Use same all-open configuration for generated Kotlin Maven projects as Gradle

geoand committed:
Ensure that the Kotlin compiler adds method param names to bytecode

We already do this for Java, so let's make it consistent for Kotlin too

Fixes: #6041

gastaldi committed:
Added back methods mistakenly considered unused in QuarkusPluginExtension

These methods were removed in 3265edd082955d6f5b0498dfeb79033c2c073219 because they were considered unused but they are used in the Kotlin Gradle build.
Added tests for all SourceTypes now

Fixes #6311

gastaldi committed:
Show stacktrace when building the gradle plugin

gastaldi committed:
Disabling Scala gradle build test for now

gastaldi committed:
Fixes Illegal char error in Gradle plugin

gastaldi committed:
Revert "Disabling Scala gradle build test for now"

This reverts commit b3ea880dabcbbeb135f5d47879e91838aa11016a.

gastaldi committed:
Set className and path to trigger sample code generation

gastaldi committed:
basic-rest/scala/build.gradle-template.ftl requires scala-library

Fixes #6316

geoand committed:
Ensure that Kotlin Data classes with default values work in native with JAX-RS

Fixes: #3954

gsmet committed:
Add missing descriptions to recently added extensions

gsmet committed:
Tweak the Sentry extension description

gsmet committed:
Fix a source extract in Qute documentation

gsmet committed:
Be a bit more consistent in Qute documentation indentation

gsmet committed:
Fix a minor indentation issue in Gradle tooling documentation

gsmet committed:
Review the Logging to Sentry documentation

lordofthejars committed:
Fixes spaces to be double spaces in all the examples

rquinio committed:
Handle javax.inject.Named on implict singleton bean in spring-di

kdnakt committed:
fix typo

loicmathieu committed:
Fix Elytron properties perf issue

loicmathieu committed:
Fix Elytron JDBC perf issue

ianwormsbecker-ssimwave committed:
Update docs with import for RegisterForReflection

machi1990 committed:
docs(qute): add link to Qute reference guide in the Qute templating engine guide

Fixes https://github.com/quarkusio/quarkus/issues/6367

kdnakt committed:
docs: fix project artifact id for jackson

kdnakt committed:
fix project name in the link to legumes.html

devauxbr committed:
enable hot reload on application.yaml

geoand committed:
Create BuildTimeRunTimeDefaultValuesConfigSource only once

geoand commented:
I think we should also have https://github.com/quarkusio/quarkus/pull/6263 in master and this as well :)

gsmet commented:
@geoand yes, it's already marked. I'll backport it once it lands into master.

I have a few others I marked too, this is not the final PR.

geoand commented:
Great, thanks for the information @gsmet!

geoand committed:
Document how to retrieve the active profile

loicmathieu committed:
Add JSON-B and Jackson serialization to the Kafka guide

nmcl committed:
Improved STM text

Will make further changes once I've modified the quickstart code itself.

https://github.com/quarkusio/quarkusio.github.io/issues/411

https://github.com/quarkusio/quarkusio.github.io/pull/417

devauxbr committed:
add application.yaml hot reload automated test

gunnarmorling committed:
Typo fix in Neo4j guide

gsmet committed:
Upgrade Jackson BOM to 2.9.10.20200103

gsmet committed:
Take into account the web.xml MIME mappings in the Undertow extension

Fixes #6399

michael-simons committed:
Improve the Neo4j guide.

The Neo4j driver has been updated to 4.0.0 final and Neo4j 4.0 can be
used from Docker, the preview flag can now be removed.

The final driver revealed some errors in the existing guide:

The reading of the transactional result of a transactional function
needs to happen inside the same transaction.
Therefor the guide needs a fix so that the composition of the future
extracting the node happens in the transaction, not outside.

geoand committed:
Fix Kubernetes manifest port handling

This is needed because of the config changes that went in 1.1.
One thing that needed to be removed because of those changes was
the warning we used to produce about potential mismatch between
the port in the Kubernetes manifests and the port set at runtime

Fixes: #6408

ia3andy committed:
Make quarkus.log.sentry.dsn required only when sentry is enabled

gsmet committed:
Check that there is a pluginOptions child before iterating on it

Fixes #6428
